src/components/booking/PaymentStep.tsx

```
@@ -21,59 +22,251 @@ import {
+      initializeStripe();
+    }
+
+    // Cleanup on unmount or payment method change
+    return () => {
+      if (cardElement) {
+        cardElement.unmount();
+        setCardElement(null);
+        setStripeElements(null);
+      }
+    };
+  }, [bookingData.paymentMethod, paymentConfig?.stripe?.enabled]);
```

Biome (1.9.4)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)